patterns/testimonials-3-columns.php

```diff
@@ -6,24 +6,33 @@
  */
 
 use Automattic\WooCommerce\Blocks\Patterns\PatternsHelper;
+
 $content = PatternsHelper::get_pattern_content( 'woocommerce-blocks/testimonials-3-columns' );
+
+$main_header        = $content['titles'][0]['default'] ?? '';
```

Comment:
@nefeline I think each one of these variables should be `$content['titles'][0]['default']` without the `?? ''` condition. Otherwise, we'll potentially get empty content. Even if the AI text generation fails, we'll always have the default value there.

Reply:
Unfortunately, that's not always the case and the lack of a fallback can trigger PHP warnings. Let me try to explain this further since I may not have been super clear in the description for this PR:

- User updated the content in patterns using AI: now all content is fetched from the  `'wc_blocks_patterns_content'` option.
- We make a change in our patterns and add a new title to it in the patterns dictionary (in the `dictionary.json` file) & update the reference on the pattern itself.
- User with AI-generated content updates the Woo Blocks plugin and is now immediately greeted with a PHP warning of undefined index, since AI-content was never generated for the newly introduced title in the pattern (the DB is not in sync with the patterns dictionary).

In the long term I think we should have a scheduled action taking care of this whenever the plugin is updated, but either way, we have to be defensive on the pattern level as well to prevent this.

Reply:
Oh, I see, I didn't think about that case 🙏 
I think we need to update all other patterns as well then.

Reply:
> I think we need to update all other patterns as well then.
yep! I'm creating an issue for this later today: updating all patterns and creating the scheduled action that is triggered on plugin update to prevent this from happening.

Reply:
created the issue over here: https://github.com/woocommerce/woocommerce-blocks/issues/11166